migration/migrator/migrations/course/20211006221014_leaderboard.py

```diff
@@ -0,0 +1,55 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+    """
+        CREATE TABLE IF NOT EXISTS autograding_metrics (
+            user_id text,
+            team_id text,
+            g_id text NOT NULL,
+            g_version integer NOT NULL,
+            testcase_id text NOT NULL,
+            elapsed_time real,
+            max_rss_size integer,
+            points integer NOT NULL,
+            passed boolean NOT NULL,
+            hidden boolean NOT NULL,
+            primary key (user_id, team_id, g_id, testcase_id, g_version),
+            CONSTRAINT metrics_user_team_id_check CHECK (((user_id IS NOT NULL) OR (team_id IS NOT NULL))),
+            constraint elapsed_time_nonnegative check (elapsed_time >= 0),
+            constraint max_rss_size_nonnegative check (max_rss_size >= 0)
+        );
+    """
+    )
+
+    database.execute("ALTER TABLE electronic_gradeable_version ADD COLUMN IF NOT EXISTS anonymous_leaderboard boolean default true not null")
+
+
+def down(config, database, semester, course):
+    """
+    Run down migration (rollback).
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("DROP TABLE IF EXISTS autograding_metrics;")
+    database.execute("ALTER TABLE electronic_gradeable_version DROP COLUMN IF EXISTS anonymous_leaderboard")
```

Comment:
If we have run the contest (with all its expensive grading), but I have to rollback to an earlier version temporarily because of an unrelated bug...  I don't want to delete all of the contest data.  It's not simple or quick to "just regrade" the contest. 

Please refactor to not drop the table. 

Thanks.

Reply:
I have removed the code for the down migration so that it now does nothing. The code is commented out with the idea that if we ever need it, it is there and can be run manually or uncommented out.

https://github.com/Submitty/Submitty/pull/7216/commits/07e2a4b37e14d1ba165563754acc42341f500fb4